### PR TITLE
PLATFORM-8725 | Use recursive parser calls to preserve template dom cache

### PR DESCRIPTION
**Pull request by wladekb**

### CODE REVIEW

File: includes/LST.php [hunk 1]
```diff
@@ -888,4 +888,24 @@ public static function spaceOrUnderscore( $pattern ) {
 		// returns a pettern that matches underscores as well as spaces
 		return str_replace( ' ', '[ _]', $pattern );
 	}
+
+	/**
```
Comment (mszabo-wikia) [inline includes/LST.php:892]: Let's add quick description here about the benefits

File: includes/LST.php [hunk 2]
```diff
@@ -888,4 +888,24 @@ public static function spaceOrUnderscore( $pattern ) {
...
+	 * @param \ParserOptions $options
+	 * @return string
+	 */
+	protected static function callParserPreprocess( Parser $parser, $text, $page, $options ): string {
+		global $wgDplUseRecursivePreprocess;
+		if ( $wgDplUseRecursivePreprocess ) {
+			$outputType = $parser->getOutputType();
+			$parser->setOutputType( OT_PREPROCESS );
+			$text = $parser->recursivePreprocess( $text );
+			$parser->setOutputType( $outputType );
```
Comment (mszabo-wikia) [inline includes/LST.php:905]: does regular preprocess() reset output type?

Comment (wladekb) [reply includes/LST.php]: good catch

File: includes/LST.php (hunk 1 above)
Comment (wladekb) [reply includes/LST.php:892]: have a look at an elaborative essay now :)

**Review by mszabo-wikia — APPROVED** [review]

**Pull request merged**